### ISSUE TITLE
Variable selection does not improve RMSE

### ISSUE THREAD

**Issue by JRSlotman** [dhopp1/nowcastLSTM#11]
I've experimented with the variable selection function to reduce the feature space and keep only those features with (hopefully) the most explanatory power. However, I noticed the number of features that is selected by the function is very low and the selected features don't seem to be the most informative. This is confirmed by a more thorough test I ran to compare RMSE on a test set after training the model many times and for multiple countries on 1. only the suggested features or 2. the full set of features. RMSE varied of course but was about twice as high after variable selection. I also don't get anywhere close to the performance calculated by the variable_selection function, even when calculating RMSE on the train data. In fact, RMSE What am I doing wrong in my setup? 

Specifically, I take the following steps:
1. Create train data with cutoff that leaves a few years (~20%) of test data;
2. Feature engineering on train data (imputation, remove near-zero variation and highly correlated features, centering and scaling);
3. Find tune variables using variable_selection with identical parameters* as the model (n_timesteps=12, train_episodes = 5, batch_size = 64, decay = 0.995, n_hidden = 5, n_layers = 2, dropout = 0.23, optimizer = "torch.optim.RMSprop", criterion = "torch.nn.MSELoss()") and initial_ordering = "feature_contribution";
4. Create copy of traindata using only date, target variable and tune variables;
5. Train model on new traindata;
6. LSTM predict on testdata (only_actuals_obs = TRUE);
7. Calculate RMSE on observations in testdata after the train cutoff date 

* The hyperparameters are tuned mostly for speed, not accuracy.

**Comment by dhopp1** [comment]
Interesting, this is probably easier to clear up with a quick call. Shoot me an email and we can set up something for next week maybe? daniel.hopp1@gmail.com
